include/nbl/video/IGPURayTracingPipeline.h

```diff
@@ -0,0 +1,107 @@
+#ifndef _NBL_I_GPU_RAY_TRACING_PIPELINE_H_INCLUDED_
+#define _NBL_I_GPU_RAY_TRACING_PIPELINE_H_INCLUDED_
+
+#include "nbl/asset/IPipeline.h"
+#include "nbl/asset/IRayTracingPipeline.h"
+
+#include "nbl/video/SPipelineCreationParams.h"
+
+
+namespace nbl::video
+{
+
+class IGPURayTracingPipeline : public IBackendObject, public asset::IRayTracingPipeline<const IGPUPipelineLayout, const IGPUShader>
+{
+        using pipeline_t = asset::IRayTracingPipeline<const IGPUPipelineLayout,const IGPUShader>;
+
+    public:
+
+        struct SShaderGroupHandle
+        {
+          private:
+            uint8_t data[video::SPhysicalDeviceLimits::ShaderGroupHandleSize];
+        };
+        static_assert(sizeof(SShaderGroupHandle) == video::SPhysicalDeviceLimits::ShaderGroupHandleSize);
+
+        struct SCreationParams final : pipeline_t::SCreationParams, SPipelineCreationParams<const IGPURayTracingPipeline>
+        {
+            #define base_flag(F) static_cast<uint64_t>(pipeline_t::SCreationParams::FLAGS::F)
+            enum class FLAGS : uint64_t
+            {
+                NONE = base_flag(NONE),
+                DISABLE_OPTIMIZATIONS = base_flag(DISABLE_OPTIMIZATIONS),
+                ALLOW_DERIVATIVES = base_flag(ALLOW_DERIVATIVES),
+                CAPTURE_STATISTICS = base_flag(CAPTURE_STATISTICS),
+                CAPTURE_INTERNAL_REPRESENTATIONS = base_flag(CAPTURE_INTERNAL_REPRESENTATIONS),
+                FAIL_ON_PIPELINE_COMPILE_REQUIRED = base_flag(FAIL_ON_PIPELINE_COMPILE_REQUIRED),
+                EARLY_RETURN_ON_FAILURE = base_flag(EARLY_RETURN_ON_FAILURE),
+                LINK_TIME_OPTIMIZATION = base_flag(LINK_TIME_OPTIMIZATION),
+                RETAIN_LINK_TIME_OPTIMIZATION_INFO_BIT_EXT = base_flag(RETAIN_LINK_TIME_OPTIMIZATION_INFO),
+                RAY_TRACING_SKIP_TRIANGLES_BIT_KHR = 1<<12,
+                RAY_TRACING_SKIP_AABBS_BIT_KHR = 1<<13,
```

Comment:
IMHO these should be flags in `IRayTracingPipeline`, the CPU asset will need to know about them

Reply:
Also:
1. lets not use `KHR`, `EXT` or `NV` suffices
2. avoid tautologies (DRY) the scope is already `IRayTracingPipeline::SCreationParams::FLAGS` we don't need another `RAY_TRACING` in the name
3. lets call `SKIP_TRIANGLES_BIT_KHR` `SKIP_BUILT_IN_PRIMITIVES_BIT` instead, in line with Vk 1.4 and maintenance extensions (they're the same value)

Reply:
> lets call SKIP_TRIANGLES_BIT_KHR SKIP_BUILT_IN_PRIMITIVES_BIT instead, in line with Vk 1.4 and maintenance extensions (they're the same value)

this is still outstanding